core/native/src/main/scala/zio/ClockPlatformSpecific.scala

```diff
@@ -52,102 +53,53 @@ private[zio] trait ClockPlatformSpecific {
-      val ptrOfPtr = handle.asInstanceOf[Ptr[Ptr[Byte]]]
-      val dataPtr  = !ptrOfPtr
-      if (dataPtr == null) null.asInstanceOf[T]
-      else {
-        val rawptr = toRawPtr(dataPtr)
-        castRawPtrToObject(rawptr).asInstanceOf[T]
-      }
-    }
-    @alwaysinline def setData(handle: Ptr[Byte], obj: Object): Unit = {
-      // data is the first member of uv_loop_t
-      val ptrOfPtr = handle.asInstanceOf[Ptr[Ptr[Byte]]]
-      if (obj != null) {
-        if (references.contains(obj)) references(obj) += 1
-        else references(obj) = 1
-        val rawptr = castObjectToRawPtr(obj)
-        !ptrOfPtr = fromRawPtr[Byte](rawptr)
-      } else {
-        !ptrOfPtr = null
-      }
-    }
-    private val onCloseCB: CloseCB = (handle: UVHandle) => {
-      stdlib.free(handle)
-    }
+  // Multi-threaded scheduler using ScheduledExecutorService
+  val scheduler: ScheduledExecutorService = Executors.newScheduledThreadPool(4)
 
-    @alwaysinline def close(handle: Ptr[Byte]): Unit =
-      if (getData(handle) != null) {
-        uv_close(handle, onCloseCB)
-        val data    = getData[Object](handle)
-        val current = references(data)
-        if (current > 1) references(data) -= 1
-        else references.remove(data)
-        setData(handle, null)
-      }
-  }
-
-  @alwaysinline final class Timer private (private val ptr: Ptr[Byte]) extends AnyVal {
-    def clear(): Unit = {
-      uv_timer_stop(ptr)
-      HandleUtils.close(ptr)
-    }
+  final class Timer private (private val scheduledFuture: java.util.concurrent.ScheduledFuture[_]) extends AnyVal {
+    def clear(): Unit =
+      scheduledFuture.cancel(false)
   }
 
   object Timer {
-    import scala.scalanative.loop.LibUV._
-    private val timeoutCB: TimerCB = (handle: TimerHandle) => {
-      val callback = HandleUtils.getData[() => Unit](handle)
-      callback.apply()
-    }
 
-    @alwaysinline private def startTimer(
-      timeout: Long,
-      repeat: Long,
-      callback: () => Unit
-    ): Timer = {
-      val timerHandle = stdlib.malloc(uv_handle_size(UV_TIMER_T))
-      uv_timer_init(EventLoop.loop, timerHandle)
-      HandleUtils.setData(timerHandle, callback)
-      val timer = new Timer(timerHandle)
-      uv_timer_start(timerHandle, timeoutCB, timeout, repeat)
-      timer
+    def delay(duration: FiniteDuration)(implicit trace: Trace): UIO[Unit] =
+      for {
+        promise <- Promise.make[Nothing, Unit]
+        _       <- ZIO.succeed(timeoutWithTrace(duration)(() => promise.succeed(()).unit))
+        _       <- promise.await
+      } yield ()
+
+    // New method added with Trace parameter for new functionality
+    private def timeoutWithTrace(duration: FiniteDuration)(callback: () => Unit)(implicit trace: Trace): Timer = {
+      val scheduledFuture = scheduler.schedule(
+        new Runnable {
+          override def run(): Unit = callback()
+        },
+        duration.toMillis,
+        TimeUnit.MILLISECONDS
```

Comment:
Would it make sense to have higher precision, like Microseconds?

Reply:
Yes it makes sence of increasing the precision of time units to microseconds can make more precise in scheduling tasks